addons/account_payment/controllers/portal.py

```diff
@@ -55,15 +60,25 @@ def _invoice_get_page_view_values(self, invoice, access_token, **kwargs):
                 providers_sudo
             ),
         }
+        installment = invoice.get_next_installment_due()
+        next_installment = (
+            installment
+            and installment['type'] != 'early_payment_discount'
+            and installment['amount_residual_currency_unsigned'] != invoice.amount_residual
+        )
         payment_context = {
             'amount': invoice.amount_residual,
+            'amount_custom': float(kwargs['amount']) if kwargs.get('amount') else False,
+            'amount_next_installment': installment['amount_residual_currency_unsigned'] if next_installment else False,
+            'number_next_installment': installment['number'] if next_installment else False,
```

Comment:
`False` for amounts and numbers?

Reply:
I just check whether it's falsy ...

Reply:
It works for now for sure, but it'd be more logical and robust to match the type so that someone else doesn't end up adding `if (this.amount >== this.paymentContext['amount_custom'])` by mistake in 6 months.

Reply:
This has been changed completely and I'm not using the same type as default values as well.